src/mlio/default_instance_reader.cxx

```diff
@@ -158,20 +161,58 @@ default_instance_reader::handle_nested_errors()
 std::optional<memory_slice>
 default_instance_reader::read_record_payload()
 {
+    std::string split_record_err_str = "Split record is not formatted with the correct cflag.";
```

Comment:
You are unnecessarily initializing an `std::string` object on the heap every time this function is called. You also don't need an `std::string` object actually. The exception type constructors expect a `char const *` not an `std::string`. The line below is the most straightforward way to initialize the exception.

```
throw corrupt_record_error{"..."};
```

Reply:
Can I just create a char const * in the header file and reuse the error message? I didn't want to write the same raw string every time.